### PR TITLE
native version dpnp.any()

### PR DESCRIPTION
**Pull request by Rubtsowa**
tests/test_logic.py::test_any[(4,)-float64]
tests/test_logic.py::test_any[(4,)-float32]
tests/test_logic.py::test_any[(4,)-int64]
tests/test_logic.py::test_any[(4,)-int32]
tests/test_logic.py::test_any[(4,)-bool] 
tests/test_logic.py::test_any[(4,)-bool_]
tests/test_logic.py::test_any[(2,3)-float64]
tests/test_logic.py::test_any[(2,3)-float32]
tests/test_logic.py::test_any[(2,3)-int64]
tests/test_logic.py::test_any[(2,3)-int32]
tests/test_logic.py::test_any[(2,3)-bool] 
tests/test_logic.py::test_any[(2,3)-bool_]
tests/test_logic.py::test_any[(2,2,2)-float64]
tests/test_logic.py::test_any[(2,2,2)-float32]
tests/test_logic.py::test_any[(2,2,2)-int64] 
tests/test_logic.py::test_any[(2,2,2)-int32] 
tests/test_logic.py::test_any[(2,2,2)-bool] 
tests/test_logic.py::test_any[(2,2,2)-bool_]

### CODE REVIEW

File: dpnp/backend/kernels/dpnp_krnl_logic.cpp [hunk 1]
```diff
@@ -52,6 +52,28 @@ void dpnp_all_c(void* array1_in, void* result1, const size_t size)
     return;
 }
 
+template <typename _DataType, typename _ResultType>
+void dpnp_any_c(void* array1_in, void* result1, const size_t size)
+{
+    _DataType* array_in = reinterpret_cast<_DataType*>(array1_in);
```
Comment (samir-nasibli) [inline dpnp/backend/kernels/dpnp_krnl_logic.cpp:58]:
```suggestion
void dpnp_any_c(const void* array1_in, void* result1, const size_t size)
{
    const _DataType* array_in = reinterpret_cast<const _DataType*>(array1_in);
```
it's better have const pointer for input array.

File: dpnp/backend/kernels/dpnp_krnl_logic.cpp [hunk 2]
```diff
@@ -52,6 +52,28 @@ void dpnp_all_c(void* array1_in, void* result1, const size_t size)
...
+    _ResultType* result = reinterpret_cast<_ResultType*>(result1);
+
+    bool res = false;
```
Comment (samir-nasibli) [inline dpnp/backend/kernels/dpnp_krnl_logic.cpp:61]: add, please, case if you have `nullptrs`

File: dpnp/backend/kernels/dpnp_krnl_logic.cpp [hunk 3]
```diff
@@ -52,6 +52,28 @@ void dpnp_all_c(void* array1_in, void* result1, const size_t size)
     return;
 }
 
+template <typename _DataType, typename _ResultType>
```
Comment (samir-nasibli) [inline dpnp/backend/kernels/dpnp_krnl_logic.cpp:55]: again, do not understand why do you templating output type? You have instances only for `bool`.

**Review by samir-nasibli — COMMENTED** [review]
Again, no info about passed/disabled dpnp own tests and numpy original tests.
Development without test coverage is bad practice.

File: dpnp/backend/kernels/dpnp_krnl_logic.cpp (hunk 3 above)
Comment (samir-nasibli) [reply dpnp/backend/kernels/dpnp_krnl_logic.cpp:55]: I did a little research myself.
And I found out that there is an `out` param which can be different from `np.boolean`:
See:
```python
>>> o=np.array(False).astype(np.uint8)
>>> z=np.any([-1, 4, 5], out=o)
>>> z
array(1, dtype=uint8)
>>> id(z)
140070830227248
>>> id(o)
140070830227248
```
Therefore, we can save the templating output for future implementation and enabling `out` param.

**Comment by samir-nasibli** [comment]
> tests/test_logic.py::test_any[(4,)-float64]
> tests/test_logic.py::test_any[(4,)-float32]
> tests/test_logic.py::test_any[(4,)-int64]
> tests/test_logic.py::test_any[(4,)-int32]
> tests/test_logic.py::test_any[(4,)-bool]
> tests/test_logic.py::test_any[(4,)-bool_]
> tests/test_logic.py::test_any[(2,3)-float64]
> tests/test_logic.py::test_any[(2,3)-float32]
> tests/test_logic.py::test_any[(2,3)-int64]
> tests/test_logic.py::test_any[(2,3)-int32]
> tests/test_logic.py::test_any[(2,3)-bool]
> tests/test_logic.py::test_any[(2,3)-bool_]
> tests/test_logic.py::test_any[(2,2,2)-float64]
> tests/test_logic.py::test_any[(2,2,2)-float32]
> tests/test_logic.py::test_any[(2,2,2)-int64]
> tests/test_logic.py::test_any[(2,2,2)-int32]
> tests/test_logic.py::test_any[(2,2,2)-bool]
> tests/test_logic.py::test_any[(2,2,2)-bool_]

Thank you! @Rubtsowa
Nice job 👍 

And what about `numpy` original ones?

**Review by densmirn — COMMENTED** [review]
Now we are able to support parameters `axis` and `out`. Wouldn't we like to do that in this PR?

**Review by shssf — APPROVED** [review]

**Pull request merged**